akka-http-core/src/main/scala/akka/http/scaladsl/OutgoingConnectionBuilder.scala

```diff
@@ -54,6 +55,17 @@ trait OutgoingConnectionBuilder {
    */
   def http2(): Flow[HttpRequest, HttpResponse, Future[OutgoingConnection]]
 
+  /**
+   * Create a flow that when materialized creates a managed HTTP/2 TLS connection with a default port 443.
+   *
+   * The connection will be re-established as needed.
+   *
+   * Note that the responses are not guaranteed to arrive in the same order as the requests go out (In the case of a HTTP/2 connection)
+   * so therefore requests needs to have a [[akka.http.scaladsl.model.RequestResponseAssociation]]
+   * which Akka HTTP will carry over to the corresponding response for a request.
+   */
+  def managedPersistentHttp2(): Flow[HttpRequest, HttpResponse, NotUsed] // FIXME: provide some API
```

Comment:
> FIXME: provide some API

Do you mean an API to avoid the explosion of methods we have?

Reply:
Good question... :) I'll try to figure out what I thought.

Reply:
> Do you mean an API to avoid the explosion of methods we have?

Regarding the explosion, we could also introduce a flag to make any of the other flows managed as well (or at least the tested and documented variants, i.e. HTTP/2 ones for now).

Reply:
Hard to do right now because the managed variants don't have a materialized value while the regular ones offer information about the concrete connection.